src/main/java/net/minecraftforge/event/terraingen/DecorateBiomeEvent.java

```diff
@@ -96,27 +98,94 @@ public Post(World world, Random rand, BlockPos pos)
 
     /**
      * This event is fired when a chunk is decorated with a biome feature.
-     *
+     * The amount denotes how many instances of the feature will be generated.
+     * <p>
      * You can set the result to DENY to prevent the default biome decoration.
      */
     @HasResult
     public static class Decorate extends DecorateBiomeEvent
     {
+        private final EventType type;
+        @Nullable
+        private final Integer totalAmount;
+        @Nullable
+        private Integer modifiedAmount;
+
```

Comment:
Please again read the use case statement of the first comment. These are absolutely necessary for partial replacement of decoration types, and the absense of these have caused issues between several mods already, as many biome mods replace the default biome generator algorithm entirely.